tests.md

```diff
@@ -0,0 +1,833 @@
+  describe "HW_WALLETS_03 - Cannot do operations requiring private key"
+        it "Cannot send tx"
+        it "Cannot update pass"
+  describe "HW_WALLETS_04 - Can manage HW wallet the same way as others"
+        it "Can update name"
+        it "Can get tx fee"
+        it "Can delete"
+        it "Can see utxo"
+        it "Can list addresses"
+        it "Can have address pool gap"
+        it "Can list transactions"
+  describe "HW_WALLETS_05 - Wallet from pubKey is available"
+        it "The same account and mnemonic wallet can live side-by-side"
+  describe "HW_WALLETS_06 - Test parameters"
+    describe "Wallet names valid"
+    describe "Wallet names invalid"
+    describe "Pub Key invalid"
+    describe "Address pool gap invalid"
+
+spec = describe "BYRON_HW_WALLETS"
+    it "HW_WALLETS_01 - Restoration from account public key preserves funds"
+  describe "HW_WALLETS_03 - Cannot do operations requiring private key"
+        it "Cannot send tx"
+        it "Cannot update pass"
+  describe "HW_WALLETS_04 - Can manage HW wallet the same way as others"
+        it "Can update name"
+        it "Can get tx fee"
+        it "Can delete"
+        it "Can see utxo"
+        it "Can list addresses"
+        it "Can have address pool gap"
+        it "Can list transactions"
+        it "Can get coin selection"
+  describe "HW_WALLETS_05 - Wallet from pubKey is available"
+        it "Can get wallet"
+        it "Can list wallet"
+        it "The same account and mnemonic wallet can live side-by-side"
+
+spec = describe "NEW_SHELLEY_TRANSACTIONS"
+    it "TRANS_NEW_JOIN_01b - Invalid pool id"
+    it "TRANS_NEW_JOIN_01b - Absent pool id"
+    it "TRANS_NEW_JOIN_01c - Multidelegation not supported"
+    it "TRANS_NEW_JOIN_01d - Multiaccount not supported"
+    it "TRANS_NEW_JOIN_01e - Can re-join and withdraw at once"
+    it "TRANS_NEW_JOIN_01f - Cannot re-join the same pool in Babbage"
+    it "TRANS_NEW_JOIN_01f - Can re-join the same pool in Conway"
+    it "TRANS_NEW_QUIT_01 - Cannot quit if not joined"
+```
+
+## E2E Tests
+
+the following tests are actually not about cardano-wallet at all but testing other utilities: [cardano-address]()
+They also depend on code defined in [cardano-wallet-rb](https://github.com/piotr-iohk/cardano-wallet-rb)
+
+**They can be safely discarded**
+
+```
+misc:
+     describe CardanoWallet::Misc::Node do
+       it 'Can get latest block header info' do
+     describe CardanoWallet::Misc::Network do
+       it 'Can get network information' do
+       it 'Can check network clock offset' do
+       it 'Can check network parameters' do
+         it 'SMASH health - unreachable' do
+         it 'SMASH health - bad url' do
+       it 'Inspect invalid address' do
+       it 'Inspect Shelley payment address' do
+       it 'Inspect Shelley stake address' do
+       it 'Inspect Byron Random address' do
+       it 'Inspect Byron Icarus address' do
```

Comment:
It appears that these tests exercise the `GET /addresses/{addressId}` endpoint of `cardano-wallet`.